python/GangaDirac/Lib/Credentials/DiracProxy.py

```diff
@@ -0,0 +1,112 @@
+from __future__ import absolute_import
+
+import os
+import subprocess
+from datetime import datetime, timedelta
+
+import Ganga.Utility.logging
+from Ganga.Core import GangaValueError
+from Ganga.Utility.Config import getConfig
+from Ganga.GPIDev.Schema import SimpleItem
+
+from Ganga.GPIDev.Credentials.ICredentialInfo import cache
+from Ganga.GPIDev.Credentials.ICredentialRequirement import ICredentialRequirement
+from Ganga.Core.exceptions import CredentialRenewalError
+from Ganga.GPIDev.Credentials.VomsProxy import VomsProxyInfo
+
+from GangaDirac.Lib.Utilities.DiracUtilities import getDiracEnv
+
+logger = Ganga.Utility.logging.getLogger()
+
+
+class DiracProxyInfo(VomsProxyInfo):
+    """
+    A wrapper around a DIRAC proxy file
+    """
+
+    def __init__(self, requirements, check_file=False, create=False):
+        self._shell = None
+
+        super(DiracProxyInfo, self).__init__(requirements, check_file, create)
+
+    def create(self):
+        """
+        Creates the grid proxy.
+
+        Raises:
+            CredentialRenewalError: If the renewal process returns a non-zero value
+        """
+        group_command = ''
+        logger.debug('require ' + self.initial_requirements.group)
+        if self.initial_requirements.group:
+            group_command = '--group %s --VOMS' % self.initial_requirements.group
+        command = 'dirac-proxy-init --strict --out "%s" %s' % (self.location, group_command)
+        logger.debug(command)
+        self.shell.env['X509_USER_PROXY'] = self.location
+        try:
+            self.shell.check_call(command)
+        except subprocess.CalledProcessError:
+            raise CredentialRenewalError('Failed to create DIRAC proxy')
+        else:
+            logger.debug('Grid proxy {path} created. Valid for {time}'.format(path=self.location, time=self.time_left()))
+
+    @property
+    def shell(self):
+        if self._shell is None:
+            self._shell = super(DiracProxyInfo, self).shell
+            self._shell.env.update(getDiracEnv())
+        return self._shell
+
+    def destroy(self):
+        if os.path.isfile(self.location):
+            os.remove(self.location)
+
+    @cache
+    def info(self):
+        self.shell.env['X509_USER_PROXY'] = self.location
+        status, output, message = self.shell.cmd1('dirac-proxy-info --file "%s"' % self.location)
+        return output
+
+    @property
+    @cache
+    def identity(self):
+        return self.field('identity')
+
+    @property
+    @cache
+    def group(self):
+        return self.field('DIRAC group')
+
+    @cache
+    def expiry_time(self):
+        status, output, message = self.shell.cmd1('voms-proxy-info -file "%s" -timeleft' % self.location)
+        if status != 0:
+            return datetime.now()
+        return datetime.now() + timedelta(seconds=int(output))
+
+    def default_location(self):
+        return os.getenv('X509_USER_PROXY') or '/tmp/x509up_u' + str(os.getuid())
```

Comment:
Can we put this in the proper tmp directory which is defined differently on different systems and allows for `/tmp/<username>` on systems like lxplus. This means standard tools don't pick up a proxy by default and there are problems with users knowing which file is theirs given the random name. Also `uuid.uuid4` means there is a smaller chance of collision between users from what I've read but this becomes mitigated if the file is placed in the correct user's temp folder.